packages/itwinui-css/src/input-container/input-container.scss

```diff
@@ -192,6 +188,20 @@
     }
   }
 
+  // Collapse padding between input/textarea and icon/button
+  > :is(input, textarea):has(+ .iui-button[data-iui-variant='borderless'], + .iui-svg-icon[data-iui-padded='true']) {
+    padding-inline-end: 0;
+  }
+
+  > :is(.iui-svg-icon[data-iui-padded='true'], .iui-button[data-iui-variant='borderless']) + :is(input, textarea) {
+    padding-inline-start: 0;
+  }
+
+  // input/textarea is not the first child and does not directly follow a borderless button or padded icon
+  > :not(.iui-svg-icon[data-iui-padded='true']):not(.iui-button[data-iui-variant='borderless']) + :is(input, textarea) {
```

Comment:
These selectors are brittle and will be really hard to maintain. If we change anything in any of those other components, all of this will break.

Should we introduce a new class and component called `InputWithDecorations.Icon`? It can wrap `<Icon padded>` and its only purpose would be to collapse this padding.

Reply:
Ohh, I see. Are you suggesting something like https://github.com/iTwin/iTwinUI/pull/1881/files/8b9ccc1c6c41a5328d8580e62b9e3e188d936602..52047f92d2e3fcb9f3edafd880b42ee1447be93c?

Reply:
Close. The `.iui-button[data-iui-variant='borderless']` selector also needs to go. It can use the same class I think, since we already have a place to put it (`InputWithDecorations.Button`).

Reply:
Sounds good, done in https://github.com/iTwin/iTwinUI/pull/1881/files/52047f92d2e3fcb9f3edafd880b42ee1447be93c..1b2f2a9bedf0fec8b1161861f3d3a117a4dd7f48.

I reused `.iui-input-flex-container-icon` in a new private component `InputFlexContainerButton`. This also simplified the selectors.